### PR TITLE
Add test extension [ECR-2634]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Use Extension to get rid of repetitive views instantiation.

---
See: https://jira.bf.local/browse/ECR-2634

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

**Comment by DENMROOT** [comment]
It seems after @dmitry-timofeev comments that this extensions adds more headache than actual benefits. Can we keep it simple at least in tests ?

**Comment by MakarovS** [comment]
@DENMROOT at the moment we have 59 `try(MemoryDb and/or Cleaner)` blocks in our tests which would be omitted if we implement this extension. It's debatable, but this extension would allow us to get rid of about 300 lines of code

@dmitry-timofeev should we also move
```
static {
    LibraryLoader.load();
  }
```
to this extension? That'd reduce boilerplate code in tests that use views

**Comment by dmitry-timofeev** [comment]
> LibraryLoader 

It is a good idea, if the library is available at extension instantiation time.

---

Regarding the possible complexity, we can't estimate it until we see. As this is a spike, the goal here is to explore this solution and then decide if it makes sense to use it. It seems that the extension model has some nice built-in things (like a hierarchical store with built-in lifecycle management of stored objects) that might possibly make the implementation quite consice.

> at the moment we have 59 try(MemoryDb and/or Cleaner) blocks in our tests which would be omitted if we implement this extension.

On top of reducing repetition and noise in _our_ tests, this extension might be an early prototype of Java Binding Testkit, i.e., an extension that we will _distribute_ to end-users to make testing of services easier.

File: exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/CoreSchemaProxyIntegrationTest.java [hunk 1]
```diff
@@ -24,15 +24,20 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
 
 @RequiresNativeLibrary
+@ExtendWith(ViewTestExtension.class)
 class CoreSchemaProxyIntegrationTest {
 
-  static {
-    LibraryLoader.load();
+  @BeforeEach
+  void setUp(MemoryDb db) {
```
[the remark this answers is not in the document]
Comment (MakarovS) [reply exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/CoreSchemaProxyIntegrationTest.java]: Yes, it's not needed, just a demonstration that we can inject parameters into `@BeforeEach` as well

File: exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/ViewTestExtension.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      if (parameterContext.getParameter().getType() == Fork.class) {
+        return memoryDb.createFork(cleaner);
+      } else {
+        return memoryDb.createSnapshot(cleaner);
+      }
+    }
+  }
+
+  private Store getStore(ExtensionContext context) {
+    return context.getStore(Namespace.create(getClass(), context.getRequiredTestMethod()));
```
[the remark this answers is not in the document]
Comment (MakarovS) [reply exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/ViewTestExtension.java]: Yes, we could do it either way

**Comment by MakarovS** [comment]
Closing, as I don't think this is needed after TestKit extension implementation - https://github.com/exonum/exonum-java-binding/pull/913.

**Pull request closed without merging**